caproto/threading/client.py

```diff
@@ -1432,8 +1457,10 @@ def _disconnected(self, *, reconnect=True):
         # Ensure that this method is idempotent.
         if self.dead.is_set():
             return
+        tags = {'their_address': self.circuit.address,
+                'our_address': self.sock.getsockname()[:2]}
```

Comment:
Tests appear to be failing as `.sock` doesn't exist.

Once you get the tests passing, I think this is very nearly good to go. Great work!